libs/dal/src/shared/interceptors/shared-cache.interceptor.ts

```diff
@@ -10,15 +10,26 @@ export function isStoreConnected(status: string | undefined) {
   return status === STORE_CONNECTED;
 }
 
+/**
+ * The data related to the messages stored by the subscriberId
+ * therefore in order to keep the stored data fresh we need to build the key with subscriberId first.
+ * @param key
+ * @param keyConfig
+ */
+function getIdentifier(key: string, keyConfig: Record<string, undefined>) {
+  return key.startsWith('Message')
+    ? keyConfig._subscriberId ?? (keyConfig.subscriberId && keyConfig._id) ?? keyConfig.id
+    : keyConfig._id ?? keyConfig.id ?? keyConfig._subscriberId ?? keyConfig.subscriberId;
+}
+
```

Comment:
So we need this part of the code only due to the reason we query the messages through the _subscriberId. Now sure how to make this logic simpler.

Reply:
Is it possible to split it to multiple variables? This section is very hard to read and understand :)

Reply:
most definitely possible! i was not sure it was worth the extra variable but now i do :)